lib/IRGen/GenClangDecl.cpp

```diff
@@ -56,6 +57,17 @@ class ClangDeclFinder
     callback(CXXCE->getConstructor());
     return true;
   }
+
+  bool VisitVarDecl(clang::VarDecl *VD) {
+    if (auto cxxRecord = VD->getType()->getAsCXXRecordDecl())
+      if (auto dtor = cxxRecord->getDestructor())
+        callback(dtor);
```

Comment:
Ah I see, so this gets to the dtors from the VarDecl's type rather than the ctors it already visited?

Reply:
Yes. The ctors aren't really important here; they weren't causing problems. We might need to destroy something that we never constructed. Think about it like this: what might cause us to call a destructor? Well, if we have an instance of a C++ type with a dtor, we will have to call its dtor. If that C++ type has bases/fields, same thing. To fix these issues, I look at all the places we might call a dtor, and make we tell clang to generate that symbol.

Reply:
Good approach. Can you add this as context in a comment on this VarDecl visitor?